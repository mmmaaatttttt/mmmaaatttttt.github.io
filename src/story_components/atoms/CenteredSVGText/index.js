import styled from "styled-components";
import PropTypes from "prop-types";

const CenteredSVGText = styled.text`
  text-anchor: middle;
  alignment-baseline: ${props => props.baseline};
  font-size: ${props => props.fontSize};
`;

CenteredSVGText.propTypes = {
  fontSize: PropTypes.string.isRequired,
  baseline: PropTypes.oneOf(["middle", "central"]).isRequired,
};

CenteredSVGText.defaultProps = {
  fontSize: "100%",
  baseline: "middle"
};

export default CenteredSVGText;
