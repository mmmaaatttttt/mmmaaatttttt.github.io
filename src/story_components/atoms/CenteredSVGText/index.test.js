import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import CenteredSVGText from ".";

it("renders successfully", () => {
  shallow(<CenteredSVGText />);
});

it("matches snapshot", () => {
  const wrapper = shallow(
    <CenteredSVGText fontSize="90%" baseline="central">
      in the center
    </CenteredSVGText>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
