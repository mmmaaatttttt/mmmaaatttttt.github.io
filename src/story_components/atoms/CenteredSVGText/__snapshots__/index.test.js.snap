// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<StyledComponent
  baseline="central"
  fontSize="90%"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  text-anchor: middle;
  alignment-baseline: ",
          [Function],
          ";
  font-size: ",
          [Function],
          ";
",
        ],
      },
      "defaultProps": Object {
        "baseline": "middle",
        "fontSize": "100%",
      },
      "displayName": "styled.text",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "baseline": [Function],
        "fontSize": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "text",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  in the center
</StyledComponent>
`;
